### ISSUE TITLE
Cisco Packet Tracer 8 install broken

### ISSUE THREAD

**Issue by doolphin** [NixOS/nixpkgs#291445]
### Describe the bug
When I install packet tracer 8.2.1 and run it, it launches, but gives a pop up error Titled, "Cisco Packet Tracer - Missing Script Module" with a description, "Required Script Modules do not appear to be running. Please reinstall Cisco Packet Tracer."

### Steps To Reproduce
1. Download CiscoPacketTracer_821_Ubuntu_64bit.deb from cisco netacad and verify its checksum.
2. Add it to your nix cache with $nix-prefetch-url --type sha256 file:///path/to/CiscoPacketTracer_821_Ubuntu_64bit.deb
3. In /etc/configuration.nix
4. Add pkgs.ciscoPacketTracer8 to package list
5. nixos-rebuild switch
6. Launch packet tracer 8

### Expected behavior
Ideally, packet tracer should launch with no issues, and the scripts (including pre-installed and necessary scripts) would work.

### Screenshots
![image](https://github.com/NixOS/nixpkgs/assets/10409356/7e1c4357-2894-409e-9acf-c5b275d0bb87)
![image](https://github.com/NixOS/nixpkgs/assets/10409356/52ab7e86-badd-46bd-9d0e-638f258141cf)

### Additional context
Not only is it unable to load the scripts, but it cannot see them. I have verified that the .pts files exist in the correct location on install, but trying to add them yields no results.

### Notify maintainers
@lucasew 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.3, NixOS, 23.11 (Tapir), 23.11.20240201.25e3d4c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by lucasew** [comment]
As far as I remember if you click OK then start building the network it works fine.

**Comment by sjcp7** [comment]
@lucasew I also came across this issue. The program generally works, except for IoT things. Whenever I try to configure an IoT registration server, the application crashes.

**Comment by doolphin** [comment]
Well I solved my actual issue, which was that i couldn't use netacad to launch connected .pta files and do assignments, the answer was the I had to use Google Chrome, rather than chromium. (I already ruled out Firefox). I'll leave this open since the script error does exist and affects IoT according to sjcp7.

**Comment by ob7** [comment]
for me the program will not stay running, it crashes when trying to login:
''packettracer8 
Fatal: Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile) 
Aborted (core dumped)
''

**Comment by Anifyuli** [comment]
I get this issue too

**Comment by Guccel** [comment]
Same issue as above running 822

**Comment by NemoSomnium** [comment]
Same issue, can also add that packet tracer crashes when trying to open a lab in physical mode. running packettracer 822

**Comment by lucasew** [comment]
I have no idea how to fix it lol

**Comment by gepbird** [comment]
TLDR: Seems like only signed script modules give this error, perhaps we are missing a p12 library?

On a non-NixOS system (arch for example) all the script modules are loaded, it looks like this:

![2024-09-15_11-12](https://github.com/user-attachments/assets/29f47110-63b6-4fb3-ad4f-e280aa7ef406)


Editing a script module (for example Clear Terminal Agent since it's the smallest), removing the signature (click on Resign and Edit then Cancel then Yes for "Without selecting a valid .p12 file would remove signing for this script module. Do you want to continue?") and exporting this file, the new script module is ~1kb smaller. But more importantly, it can be loaded on NixOS:

![2024-09-15_11-18](https://github.com/user-attachments/assets/5236d658-9878-42aa-bcae-1911d14bb4d5)

(I have also changed the version string for the script module to make debugging a little easier)

I searched for pkcs nix packages, but only found p11 related packages, not p12. I tried just putting them into `buildInputs`, but it didn't help. Please anyone who is more experienced with crypto take a look at this.

**Comment by griffi-gh** [comment]
So can we just.. forcefully un-sign these built-in modules as a workaround?

**Comment by gepbird** [comment]
> So can we just.. forcefully un-sign these built-in modules as a workaround? Still won't help with loading external signed modules though..

If you know how to remove the signature from a .pts file with CLI tools, I think that's a decent workaround. I've messed a little with `openssl pkcs12`, but couldn't get it to remove the signature.

But if we can't do it with CLI tools, in theory we could un-sign the modules using packet tracer, and redistribute the those modules. Except similarly to redistributing the .deb, I don't think redistributing the modules is allowed either.

**Comment by griffi-gh** [comment]
Trying to open PC crashes (Segfault)

**Comment by gepbird** [comment]
> Trying to open PC crashes (Segfault)

@griffi-gh 
Opening a PC in packet tracer works for me, can you share more details?

However IoT devices make PT crash

**Comment by griffi-gh** [comment]
> > Trying to open PC crashes (Segfault)
> 
> @griffi-gh Opening a PC in packet tracer works for me, can you share more details?
> 
> However IoT devices make PT crash

i have the two required plugins installed using signature removal trick
clicking on the pc makes the window appear, then the whole pt freezes and segfaults.

**Comment by gepbird** [comment]
Thanks @griffi-gh !

Yep, it segfaults for me too with an unsigned PT INTERNAL module.

I'll test previous versions too to see where this issue first happened.

I already tried to look into the binary with Ghidra/IDA, found the "Unable to deserialize" string but wasn't sure how to continue from there. Maybe someone smarter could reverse engineer it?

**Comment by cooparo** [comment]
For me doesn't even install.
When I added the package `ciscoPacketTracer8` to home-manager or configuration.nix, I get:
 ```shell
 > ***
 > Unfortunately, we cannot download file CiscoPacketTracer822_amd64_signed.deb automatically.
 > Please go to https://www.netacad.com to download it yourself, and add it to the Nix store
 > using either
 >   nix-store --add-fixed sha256 CiscoPacketTracer822_amd64_signed.deb
 > or
 >   nix-prefetch-url --type sha256 file:///path/to/CiscoPacketTracer822_amd64_signed.deb
 >
 > ***
```
then I: 
- downloaded the `Packet_Tracer822_amd64_signed.deb` file from cisco website
- addded to nix store with `nix-store --add-fixed sha256 CiscoPacketTracer822_amd64_signed.deb`, got `/nix/store/6562jz9mm4cika220bjxhbh42chp33hr-Packet_Tracer822_amd64_signed.deb`, so seems added successfully
- then, I rebuild and i get again same error "Unfortunately, we cannot download file CiscoPacketTracer822_amd64_signed.deb automatically. [...]"

Same happens for prev version of the pkgs `ciscoPacketTracer7`, or with `nix-prefetch-url --type sha256 file:///path/to/CiscoPacketTracer822_amd64_signed.deb` (instead of `nix-store --add-fixed`)

**Comment by griffi-gh** [comment]
> For me doesn't even install. When I added the package `ciscoPacketTracer8` to home-manager or configuration.nix, I get:
> 
> ```shell
> > ***
> > Unfortunately, we cannot download file CiscoPacketTracer822_amd64_signed.deb automatically.
> > Please go to https://www.netacad.com to download it yourself, and add it to the Nix store
> > using either
> >   nix-store --add-fixed sha256 CiscoPacketTracer822_amd64_signed.deb
> > or
> >   nix-prefetch-url --type sha256 file:///path/to/CiscoPacketTracer822_amd64_signed.deb
> >
> > ***
> ```
> 
> then I:
> 
>     * downloaded the `Packet_Tracer822_amd64_signed.deb` file from cisco website
> 
>     * addded to nix store with `nix-store --add-fixed sha256 CiscoPacketTracer822_amd64_signed.deb`, got `/nix/store/6562jz9mm4cika220bjxhbh42chp33hr-Packet_Tracer822_amd64_signed.deb`, so seems added successfully
> 
>     * then, I rebuild and i get again same error "Unfortunately, we cannot download file CiscoPacketTracer822_amd64_signed.deb automatically. [...]"
> 
> 
> Same happens for prev version of the pkgs `ciscoPacketTracer7`, or with `nix-prefetch-url --type sha256 file:///path/to/CiscoPacketTracer822_amd64_signed.deb` (instead of `nix-store --add-fixed`)

It's probably better to just override the source anyway, like i did here with my nixos configuration:

https://github.com/griffi-gh/nixos/blob/master/pkgs/ciscoPacketTracer8.nix

**Comment by ob7** [comment]
@cooparo 
I had a similar issue.  Finally realized the file wasn't named exactly as it is in the output.  Make sure the file name is exact before adding it to nix store.  I had an underscore in the wrong place, took me an hour to realize.

**Comment by cooparo** [comment]
@ob7 tnx u that solved my issues, my bad.

**Comment by gepbird** [comment]
Back to the original issue, I was trying earlier versions and found out packet tracer 8.0.0 segfaults on master while it's trying to verify the script modules, on later versions (tested with 8.2.0 and above) instead of segfaulting it shows the "Required script modules do not appear to be running" message box.
Backtrace of the segfault:
```
(gdb) bt
#0  0x00007ffff7d97bd8 in RSA_verify () from /nix/store/hy7smyyd58g1fwzz0p7klhblgjqzwksg-ciscoPacketTracer8-8.0.0/opt/pt/bin/libcrypto.so.1.1
#1  0x00007ffff7d95ebd in pkey_rsa_verify () from /nix/store/hy7smyyd58g1fwzz0p7klhblgjqzwksg-ciscoPacketTracer8-8.0.0/opt/pt/bin/libcrypto.so.1.1
#2  0x00007ffff7d46d9c in EVP_VerifyFinal () from /nix/store/hy7smyyd58g1fwzz0p7klhblgjqzwksg-ciscoPacketTracer8-8.0.0/opt/pt/bin/libcrypto.so.1.1
#3  0x0000555557a7f4a7 in Ipc::CCep::verifySignature(char const*, unsigned int) ()
#4  0x0000555557e88fcd in CScriptModule::deserialize(QDomNode&) ()
#5  0x0000555557ea67fd in CScriptModuleManager::readScriptModuleFromFile(QString const&) ()
#6  0x0000555557ea6cfa in CScriptModuleManager::addScriptModuleFromFile(QString const&) ()
#7  0x0000555558aa9e2d in oldMain(int, char**) ()
#8  0x0000555558ab7f5e in main ()
```

Then I checked out earlier versions of the derivation, and I found out 2920b6fc16a9ed5d51429e94238b28306ceda79e introduced this bug. On the previous commit script modules have signatures and are loaded, no error messages!

![2024-11-14_19-22](https://github.com/user-attachments/assets/631ffc06-2a75-449c-ba88-bb1c79867965)
